shopinvader_anonymous_partner/models/res_partner.py

```diff
@@ -101,3 +102,28 @@ def _get_anonymous_partner__cookie(self, cookies: Cookies):
         if not token:
             return self.env["res.partner"].sudo().browse()
         return self._get_anonymous_partner__token(token)
+
+    def _promote_from_anonymous_partner(self, anonymous_partner: ResPartner):
+        """
+        Promote an anonymous partner to an authenticated partner.
+
+        This method should be overridden by other modules to implement
+        the partner resolution logic, merging the anonymous partner cart
+        for instance.
+
+        This method can return False to prevent the anonymous partner cookie
+        from being deleted.
+        """
+
+    @api.model
+    def _promote_anonymous_partner(
+        self, partner: ResPartner, cookies: Cookies, response: Response
+    ):
+        """
+        Promote the current anonymous partner to the given authenticated partner.
+
+        This method calls the partner promotion and removes the anonymous partner cookie.
+        """
+        anonymous_partner = self._get_anonymous_partner__cookie(cookies)
+        partner._promote_from_anonymous_partner(anonymous_partner)
+        self._delete_anonymous_partner__cookie(cookies, response)
```

Comment:
If the docstring above is correct, this should be
```suggestion
        delete_cookie = partner._promote_from_anonymous_partner(anonymous_partner)
        if delete_cookie:
            self._delete_anonymous_partner__cookie(cookies, response)
```

Reply:
Nice catch, forgot to remove the docstring following https://github.com/shopinvader/odoo-shopinvader/pull/1570/commits/67e7319bd1f966bce993dfa8347a771e2544873d